core/trino-main/src/main/java/io/trino/execution/scheduler/PipelinedStageExecution.java

```diff
@@ -593,6 +593,11 @@ public boolean isFailure()
             return failureState;
         }
 
+        public boolean isRunningOrDone()
```

Comment:
Why can’t the caller just check these two conditions itself? Having a dedicated method to check for these two states doesn’t smell right.

Reply:
The caller can certainly check those two conditions itself. But I wrote the dedicated method because that's how the existing code does it. Specifically, `SqlQueryScheduler.DistributedStagesSchedulerState` has a method with the same name and the same definition

https://github.com/trinodb/trino/blob/1799d9227a4ed48508571ff1276834661114fb82/core/trino-main/src/main/java/io/trino/execution/scheduler/SqlQueryScheduler.java#L1640

and is used for the same purpose at

https://github.com/trinodb/trino/blob/1799d9227a4ed48508571ff1276834661114fb82/core/trino-main/src/main/java/io/trino/execution/scheduler/SqlQueryScheduler.java#L318

In fact, Andrii kindly pointed me to the code above and suggested me to do something similar to fix the issue. And I thought it's not a bad idea to follow the style in the existing code.

Reply:
I think It used to be that these two conditions were checked inline, but somebody suggested adding a method :-)

Reply:
@martint Should I remove `PipelinedStageExecution.isRunningOrDone` and just inline it?

Reply:
Yes, I don’t think the method adds much value. Let’s inline it.

Reply:
OK. I have removed and inlined the method.